flang/lib/Parser/message.cpp

```diff
@@ -491,7 +491,13 @@ void Messages::AttachTo(Message &msg, std::optional<Severity> severity) {
   messages_.clear();
 }
 
-bool Messages::AnyFatalError() const {
+bool Messages::AnyFatalError(bool warningsAreErrors) const {
+  if (messages_.empty()) {
```

Comment:
these six lines could be more clear as `if (warningsAreErrors{ ) return !messages_.empty(); }`